smart-metastore/src/main/java/org/smartdata/metastore/SearchableAbstractDao.java

```diff
@@ -77,7 +78,11 @@ protected List<EntityT> executeQuery(MetastoreQuery query) {
   }
 
   protected Optional<EntityT> executeSingle(MetastoreQuery query) {
-    return Optional.ofNullable(executeQuery(query))
+    return executeSingle(query, this::mapRow);
+  }
+
+  protected <T> Optional<T> executeSingle(MetastoreQuery query, RowMapper<T> rowMapper) {
+    return Optional.ofNullable(queryExecutor.execute(query, rowMapper))
```

Comment:
how about adding a limit to the query instead of fetching full list?

Reply:
Added limit. Previously, it was used only to fetch single rows using the primary key.